src/prefect/backend/flow_run.py

```diff
@@ -0,0 +1,596 @@
+                state=prefect.engine.state.Cancelled("Keyboard interrupt."),
+            )
+            logger.warning("Keyboard interrupt. Flow run cancelled, exiting...")
+        else:
+            logger.warning(
+                "Keyboard interrupt. Flow run is already finished, exiting..."
+            )
+        raise
+    except Exception as exc:
+        if not FlowRun.from_flow_run_id(flow_run_id).state.is_finished():
+            message = message.format(exc=exc.__repr__())
+            client.set_flow_run_state(
+                flow_run_id=flow_run_id, state=prefect.engine.state.Failed(message)
+            )
+            client.write_run_logs(
+                [
+                    dict(
+                        flow_run_id=flow_run_id,  # type: ignore
+                        name="prefect.backend.api.flow_run.execute_flow_run",
+                        message=message,
+                        level="ERROR",
+                    )
+                ]
+            )
+        logger.error(message, exc_info=True)
+        raise
+
+
+class FlowRun:
+    """
+    Object for getting information about a Flow Run from the backend API
+
+    Attributes:
+        flow_run_id: The uuid of the flow run
+        name: The name of the flow run
+        flow_id: The uuid of the flow this run is associated with
+        state: The state of the flow run
+        flow: Metadata for the flow this run is associated with
+        task_runs: A view of cached task run metadata associated with this flow run
+
+    """
+
+    def __init__(
+        self,
+        flow_run_id: str,
+        name: str,
+        flow_id: str,
+        state: State,
+        task_runs: Iterable["TaskRun"] = None,
+    ):
+        self.flow_run_id = flow_run_id
+        self.name = name
+        self.flow_id = flow_id
+        self.state = state
+
+        # Cached value of all task run ids for this flow run, only cached if the flow
+        # is done running
+        self._task_run_ids: Optional[List[str]] = None
+
+        # Cached value of flow metadata
+        self._flow: Optional[FlowMetadata] = None
+
+        # Store a mapping of task run ids to task runs
+        self._task_runs: Dict[str, "TaskRun"] = {}
+
+        # Store a mapping of slugs to task run ids (mapped tasks share a slug)
+        self._task_slug_to_task_run_ids: Dict[str, Set[str]] = defaultdict(set)
+
+        if task_runs is not None:
+            for task_run in task_runs:
+                self._add_task_run(task_run)
+
+    def _add_task_run(self, task_run: "TaskRun"):
+        self._task_runs[task_run.task_run_id] = task_run
+        self._task_slug_to_task_run_ids[task_run.task_slug].add(task_run.task_run_id)
+
+    def update(self) -> "FlowRun":
+        """
+        Get the a new copy of this object with the newest data from the API
+        """
+        return self.from_flow_run_id(
+            flow_run_id=self.flow_run_id,
+            load_static_tasks=False,
```

Comment:
would we want to set this to `True` here? and maybe not pass in `_task_runs`?

Could be misinterpreting here, but seems like I'd want `update` to automatically get all the newest data

Reply:
I think at the very least I should update this to only pass task runs that are in a finished state. I might even update the cache to only store task runs that are in a finished state so `get(task)` will always return new data unless the task is done.

Reply:
love that idea